tests/testmatrix.c

```diff
@@ -0,0 +1,799 @@
+#ifdef WIN32
+#ifndef __cplusplus
+#error Please compile with a C++ compiler.
+#endif
+#endif
+
+#if defined(_WIN32)
+#include <Windows.h>
+#include <GdiPlus.h>
+
+#pragma comment(lib, "gdiplus.lib")
+#else
+#include <GdiPlusFlat.h>
+#endif
+
+#ifdef WIN32
+using namespace Gdiplus;
+using namespace DllExports;
+#endif
+
+#include <assert.h>
+#include <float.h>
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+int floatsEqual(float v1, float v2)
+{
+		return fabs (v1 - v2) < 0.0001;
+}
+
+void verifyMatrix (GpMatrix *matrix, REAL e1, REAL e2, REAL e3, REAL e4, REAL e5, REAL e6)
```

Comment:
this and floatsEqual already exist in testhelpers.h, please just #include that and use it from there (and while you're at it please get rid of the copy in testpathgradientbrush.c as well).